### ISSUE TITLE
Myst III will not run with newest RetroPie 4.5

### ISSUE THREAD

**Issue by ahughes33** [residualvm/residualvm#1564]
Myst III DVD  v1.27
ResidualVM  v0.4.0git
RetroPie  v4.5
Raspberry Pi 3b+

I have succeeded in getting Grim Fandango and Escape from Monkey Island working, but Myst III absolutely will not run despite my best efforts. I've followed all the instructions: copied the Data and M3Data directories into a folder named myst3. I've renamed language.m3u to ENGLISH.m3u and placed it in the TEXT folder. I put the whole thing in the same directory as the other two working games. From the GUI I can add the game, but when I click Start the screen just goes black for about two minutes, then returns to the GUI. No error messages or anything. What is going wrong? Any help is greatly appreciated.

**Comment by Botje** [comment]
Do you have a more specific ResidualVM version number than that? Any chance you can access the residualvm log output?

**Comment by ahughes33** [comment]
Holy cow! Thanks for the quick response.  Here’s what I’ve got from the ‘About’ page in the GUI:  0.4.0git (Apr 24 2019 21:30:54)

I’m sure I could access the log output through the command interface, where should I look for it?

**Comment by Botje** [comment]
For me they were in `~/.cache/residualvm/logs/residualvm.log`. They might be in `~/.residualvm/logs` too. If you can run ResidualVM without going through the launcher, please do so. If it produces a segmentation violation or a bus error we can start debugging that.

**Comment by ahughes33** [comment]
Okay, I found the log file and it said exactly the same thing:   0.4.0git (Apr 24 2019 21:30:54)

I’m not sure if I can start ResidualVM without the launcher, but I can navigate my way around the RetroPie file structure through the command interface. If there’s an executable file that would start if from there I could do that. Otherwise I’m not sure.

**Comment by Botje** [comment]
Can you attach the rest of the log?

About launching ResidualVM by itself: simply start it from the launcher, SSH to your Pi and use `ps aux` to see what the PID and the commandline that was used for ResidualVM. If the commandline does not make it clear, look at `/proc/$pid/exe`

**Comment by ahughes33** [comment]
On Thursday, July 18, 2019, 12:23:31 PM CDT, Dries Harnie <notifications@github.com> wrote:  
 Here is the exact wording in the log, and I have attached it as a .txt file as well.
[2019-07-18 12:31:18] ResidualVM 0.4.0git (Apr 24 2019 21:30:54)[2019-07-18 12:31:18] Vorbis MP3 ALSA RGB zLib MPEG2 Theora FreeType2 JPEG PNG keymapper virtual keyboard [2019-07-18 12:31:18] --- Log opened.[2019-07-18 12:31:18] --- Log closed successfully.

I found what I believe might be the file that launches ResidualVM, it is +Start ResidualVM.sh found at /home/pi/RetroPie/roms/residualvm/+Start ResidualVM.sh    however it will not launch straight from the commandline. I may be wrong, but I think it needs to be opened within emulationstation in order to run. Please let me know if this is incorrect. Also thank you for taking the time to help me out with this.
I have attached a screenshot of the code in +Start ResidualVM.sh  just in case it helps.




 
Can you attach the rest of the log?

About launching ResidualVM by itself: simply start it from the launcher, SSH to your Pi and use ps aux to see what the PID and the commandline that was used for ResidualVM. If the commandline does not make it clear, look at /proc/$pid/exe

**Comment by Botje** [comment]
We need a log file of right after you launch myst3 and crash it. Your image attachment did not get through.

The [contents](https://github.com/RetroPie/RetroPie-Setup/blob/master/scriptmodules/emulators/residualvm.sh#L68-L80) of "Start ResidualVM.sh" seem to indicate that it just spawns a `residualvm` binary from somewhere. Please find and run that directly instead of going through the launcher.

**Comment by ahughes33** [comment]
Hello, and many many thanks for your help with this. I know you must be busy. I’m quite afraid that I don’t have the technical prowess to be able to do what you’re asking. I could be wrong, but I don’t think that this game is launchable outside of the residualVM program and without Emulationstation running. I don’t think I can type something into a linux command prompt that will make this game run. That being said, I did find this information in /dev/shm/runcommand.log right after an attempt at launching the game:

Parameters:
Executing: bash “/home/pi/RetroPie/roms/residualvm/+Start ResidualVM.sh”
Using joystick: XGaming X-Arcade 2
Virtual keyboard pack ‘vkeybd_default’ loaded successfully
WARNING: No hardware input were defined, using defaults!
User picked target ‘myst3-win’ (gameid ‘myst3’)…
   Looking for a plugin supporting this gameid… Myst III Engine
   Starting ‘Myst III Exile’
Initializing Software 3D Renderer
/home/pi/RetroPie/roms/residualvm/+Start ResidualVM.sh:  line 7:  24753 Segmentation fault    /opt/retropie/emulators/residualvm/bin/residualvm  - -renderer=$renderer   - -fullscreen   - - joystick=0  - -extrapath=“/opt/retropie/emulators/residualvm/extra”  $game

I have no idea if any of that is helpful in the least. To be honest, if this is of no use then I’m about ready to throw in the towel. Again, I cannot thank you enough for lending me your time. I greatly appreciate it, and I only wish I had enough technical know-how to utilize your input to its fullest.

All the best,

Andrew Hughes

**Comment by Botje** [comment]
"Segmentation fault" means it crashed. If you could get your hands on a core dump and a binary with debugging symbols we could actually investigate the process...

**Comment by ahughes33** [comment]
I’m good at copying and pasting, but I don’t know much about proper coding, and what you just said is way over my capabilities unfortunately. I have no idea how to get a core dump, or even what that means. You’ve been more than gracious with your time, but I am definitely the limiting factor in this equation. Maybe when RetroPie for the new Raspberry Pi 4 is released I’ll get lucky and it will work, but I think I have to face the reality that it just isn’t going to happen for me at this time. I wish I could do more, but I’ve hit my limit. Thank you so much for helping me out.

**Comment by Botje** [comment]
No problem. Thanks for taking the time in reporting!

**Comment by 3ravenous** [comment]
I had exactly the same experience as the author above, but I found an unlikely fix that matches nothing I read on the ResidualVM main repository.  I have no idea why this works, but if you add the "bin" directory from the Myst III DVD, in addition to "Data" and "M3Data", the game suddenly works. I stumbled upon the bin directory idea here (forums.residualvm.org/viewtopic.php?f=1&t=729), but there's no explanation with it. It just seemed weird, so I tried it. I can't believe it worked.
